internal/telemetry/error.go

```diff
@@ -1,19 +1,70 @@
 package telemetry
 
 import (
+	"errors"
+	"net"
+	"strings"
+
 	"github.com/getsentry/sentry-go"
+	"github.com/lib/pq"
 	"go.opentelemetry.io/otel/codes"
-	otrace "go.opentelemetry.io/otel/trace"
+	"go.opentelemetry.io/otel/trace"
+	gcodes "google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
 // RecordError should be called on a span to mark it as errored. By default error values are not recorded, unless the debug flag is set.
-func RecordError(span otrace.Span, err error, opts ...otrace.EventOption) {
+func RecordError(span trace.Span, err error, opts ...trace.EventOption) {
 	if err == nil {
 		return
 	}
 
+	if cls := classifyError(err); cls != errNoClass {
+		span.SetStatus(codes.Error, string(cls))
+		return
+	}
+
 	sentry.CaptureException(err)
 
 	span.RecordError(err, opts...)
 	span.SetStatus(codes.Error, err.Error())
 }
+
+type errClass string
+
+const (
+	errNoClass      = errClass("")
+	errCancellation = errClass("cancelled")
+	errAuth         = errClass("auth")
+	errConn         = errClass("connection")
+	errDatabase     = errClass("database")
+)
+
+func classifyError(err error) errClass {
```

Comment:
food for taught we can make this more generic i.e define a static data structure where we have for each error type possible messages that will classify / known errors etc' and then just loop over each type with the error returned to classify.

Reply:
Every error type has something to check for before it's "accepted" (as seen with the `net.OpError.Op` and `pq.Error.Code.Class()` cases) so unless it's fairly generic ("ignore all pqErrors") it would just complicate things for the sake of being able to loop through a list.